internal/manifests/targetallocator/deployment.go

```diff
@@ -24,7 +24,7 @@ import (
 )
 
 // Deployment builds the deployment for the given instance.
-func Deployment(params manifests.Params) *appsv1.Deployment {
+func Deployment(params manifests.Params) (*appsv1.Deployment, error) {
 	name := naming.TargetAllocator(params.OtelCol.Name)
 	labels := Labels(params.OtelCol, name)
```

Comment:
@swiatekm-sumo do you think we should still construct the deployment if we fail to make the configmap here? Right now we pass through, but I'm wondering if we should just reject it?

Reply:
I'd return an error. If we failed here, then we also failed to actually create the ConfigMap, in which case the target allocator won't start anyway. In that case, it's better to do nothing and bubble the error up than create faulty resources.

The best we can do here, of course, is to reject this at admission, but that's beyond the scope of this PR.